pyronear/datasets/openfire.py

```diff
@@ -118,17 +124,16 @@ def download(self, threads=None, num_samples=None):
 
         # Download actual images
         training_set, test_set = [], []
-        img_folder = self._root.joinpath(self._raw, 'images')
-        img_folder.mkdir(parents=True, exist_ok=True)
+        self.img_folder.mkdir(parents=True, exist_ok=True)
         unavailable_idxs = 0
         # Prepare URL and filenames for multi-processing
         entries = [(a['url'], a['name']) for idx, a in enumerate(annotations)]
         # Use multiple threads to speed up download
-        download_urls(entries, img_folder, threads=threads)
+        download_urls(entries, self.img_folder, threads=threads)
         # Verify downloads
         for idx, annotation in enumerate(annotations):
-            img_path = self._raw.joinpath('images', entries[idx][1])
-            if self._root.joinpath(img_path).is_file():
+            img_path = self.img_folder.joinpath(entries[idx][1])
+            if img_path.is_file():
```

Comment:
Are you positive it works when iterating over the dataset as a dataloader?
Initially `img_path` was only starting under `root` because once it's stored in the dictionary on line 140, the image will then be loaded as `root.joinpath(img_path)` if I'm not mistaking.
```
loader = torch.utils.data.DataLoader(dataset, batch_size=4, num_workers=4)
for x, target in loader:
    y = target
```
Does this work on your end? (or the pytorch training reference script for that matter)


Perhaps we should add a unittest with data loader to check this in CI

Reply:
Starting from scratch this works:
```
mkdir -p /tmp/data/OpenFire
ln -s ~/Workarea/PyroNear/data/OpenFire/images /tmp/data/OpenFire/
python references/classification/torch/train.py --device cpu --workers 0 --epochs 1 --data-path /tmp/data
```

`ls -R /tmp/data` gives:

```
OpenFire

/tmp/data/OpenFire:
images    processed raw

/tmp/data/OpenFire/processed:
test.pt     training.pt

/tmp/data/OpenFire/raw:
openfire_binary.json
```

I am not familiar with those classes but:
- iterating over the dataset works
- each element is a tuple (PIL.Image.Image, target)
- the test `is_file()` in the lines above succeeds since `train.pt` and `test.pt` are created as before.
- The one you suggest doesn't because it is an image, so: 

```
import torch, pyronear
dataset = pyronear.datasets.OpenFire('/tmp/data')
loader = torch.utils.data.DataLoader(dataset, batch_size=1, num_workers=4)

TypeError: default_collate: batch must contain tensors, numpy arrays, numbers, dicts or lists; found <class 'PIL.Image.Image'>
```

And yes, I totally agree with adding tests for dataloader, etc

Reply:
Perfect, I just wanted to make sure we don't compromise the training!
For the collate function, it requires to resize all images to the same size, but if the training is running then it's all god
Thank you very much!